chia/wallet/wallet_node.py

```diff
@@ -590,14 +608,10 @@ async def receive_state_from_peer(
         if self.validation_semaphore is None:
             self.validation_semaphore = asyncio.Semaphore(6)
 
-        # If there is a fork, we need to ensure that we roll back in trusted mode to properly handle reorgs
-        if trusted and fork_height is not None and height is not None and fork_height != height - 1:
-            await self.wallet_state_manager.reorg_rollback(fork_height)
-            await self.wallet_state_manager.blockchain.set_finished_sync_up_to(fork_height)
+        # If there is a fork, we need to ensure that we roll back to properly handle reorgs
+        if fork_height is not None and height is not None and fork_height != height - 1:
```

Comment:
wallet_short_sync_backtrack can safely rollback because we validated the weight for the new peak so we know the peer is telling the truth about the reorg.

Reply:
wallet_short_sync_backtrack can safely rollback because we validated the weight for the new peak so we know the peer is telling the truth about the reorg.